example/app/donut-chart.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useState } from "react";
+import { SafeAreaView, ScrollView, StyleSheet, View } from "react-native";
+import { LinearGradient, vec } from "@shopify/react-native-skia";
+import { Pie } from "victory-native";
+import { InfoCard } from "example/components/InfoCard";
+import { Button } from "example/components/Button";
+import { appColors } from "./consts/colors";
+import { descriptionForRoute } from "./consts/routes";
+
+function calculateGradientPoints(
+  radius: number,
+  startAngle: number,
+  endAngle: number,
+  centerX: number,
+  centerY: number,
+) {
+  // Calculate the midpoint angle of the slice for a central gradient effect
+  const midAngle = (startAngle + endAngle) / 2;
+
+  // Convert angles from degrees to radians
+  const startRad = (Math.PI / 180) * startAngle;
+  const midRad = (Math.PI / 180) * midAngle;
+
+  // Calculate start point (inner edge near the pie's center)
+  const startX = centerX + radius * 0.5 * Math.cos(startRad);
+  const startY = centerY + radius * 0.5 * Math.sin(startRad);
+
+  // Calculate end point (outer edge of the slice)
+  const endX = centerX + radius * Math.cos(midRad);
+  const endY = centerY + radius * Math.sin(midRad);
+
+  return { startX, startY, endX, endY };
+}
+
+const randomNumber = () => Math.floor(Math.random() * (50 - 25 + 1)) + 125;
+function generateRandomColor(): string {
+  // Generating a random number between 0 and 0xFFFFFF
+  const randomColor = Math.floor(Math.random() * 0xffffff);
+  // Converting the number to a hexadecimal string and padding with zeros
+  return `#${randomColor.toString(16).padStart(6, "0")}`;
+}
+
+const DATA = (numberPoints = 5) =>
+  Array.from({ length: numberPoints }, (_, index) => ({
+    value: randomNumber(),
+    color: generateRandomColor(),
+    label: `Label ${index + 1}`,
+  }));
+
+export default function DonutChart(props: { segment: string }) {
+  const description = descriptionForRoute(props.segment);
+  const [data, setData] = useState(DATA(5));
+
+  return (
+    <SafeAreaView style={styles.safeView}>
+      <ScrollView>
+        <View style={styles.chartContainer}>
+          <Pie.Chart
+            data={data}
+            labelKey={"label"}
+            valueKey={"value"}
+            colorKey={"color"}
+            innerRadius={"50%"}
+            renderLegend={(data) => (
+              <Pie.ChartLegend
+                containerStyle={{
+                  justifyContent: "center",
+                }}
+                position="bottom"
+                data={data}
```

Comment:
thoughts on putting `data` in the `PieChartProvider` so we don't have to force users to pass it here as well?

Reply:
genius